yellow-paper/docs/circuits/private-kernel-initial.md

```diff
@@ -0,0 +1,258 @@
+
+### Validating Public Inputs
+
+#### Verifying the accumulated data.
+
+It verifies that the following values align with those in the private call data:
+
+- Log hashes.
+- Log lengths.
+
+#### Verifying the transient accumulated data.
+
+1. It ensures that the following arrays match those in the private call data:
+
+   - Note hash contexts.
+   - Nullifier contexts.
+   - L2-to-L1 message contexts.
+   - New contract contexts.
+   - Read requests.
+   - Public call requests.
+
+2. It checks that the following aligns with the array in the private call data, with items arranged in **reverse** order:
+
+   - Private call requests.
+
+   > It's important that the call requests are arranged in reverse order to ensure they are executed in chronological order. This becomes particularly crucial when calling a contract deployed earlier within the same transaction.
+
+3. For the note hash contexts, it also verifies that each is associated with a nullifier counter, which is provided as a hint via the private inputs. The nullifier counter can be:
+
+   - Zero: if the note is not nullified in the same transaction.
+   - Greater than zero: if the note is nullified in the same transaction.
+     - This value must be greater than the counter of the note hash.
+
+   > Nullifier counters are used in the [reset private kernel circuit](./private-kernel-reset.md#verifying-read-requests) to ensure a read happens **before** a transient note is nullified.
+
+   > Zero can be used to indicate a non-existing transient nullifier, as this value can never serve as the counter of a nullifier. It corresponds to the _counter_start_ of the first function call.
+
+#### Verifying the constant data.
+
+It verifies that:
+
+- The transaction context matches the one in the transaction request.
+
+> The historical data must align with the data used in the private function circuit, as verified [earlier](#verifying-the-public-inputs-of-the-private-function-circuit).
+
+## Private Inputs
+
+### Transaction Request
+
+A transaction request represents the caller's intent. It contains:
+
+- Sender's address.
+- <a name="function_data">Function data</a>:
+
+  - Function selector.
+  - Function type (private/public/unconstrained).
+  - A flag indicating whether the function is an internal function.
+
+- Hash of the function arguments.
+- Transaction context
+  - A flag indicating whether it is a fee paying transaction.
+  - A flag indicating whether it is a fee rebate transaction.
+  - Chain ID.
+  - Version of the transaction.
+
+### Private Call Data
+
+The private call data holds details about the current private function call:
+
+- Contract address.
+- Function data.
```

Comment:
This comment applies for all of the bullet points that describe data types in this doc


Ideally each data type is explicitly defined somewhere. An explicit definition will contain the length of the data parameter and ideally a brief description of the type's intended purpose. E.g. the table in the "public inputs" section of `private-function.md` does this well